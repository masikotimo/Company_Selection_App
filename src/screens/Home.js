import React from 'react';
import { Text, View, Image, ScrollView } from 'react-native';
import { Card } from 'react-native-elements';
import AppIntroSlider from 'react-native-app-intro-slider';
import { slides } from '../utils/slides';
import { styles } from '../styles/IntroTourStyles';
import { Companies, inputString } from '../utils/InputFile';

const validCompanies = Companies(inputString).filter((a) => {
  if (a.isValid) {
    return true;
  }
  return false;
});
const inValidCompanies = Companies(inputString).filter((a) => {
  if (!a.isValid) {
    return true;
  }
  return false;
});

export default class Home extends React.Component {
  static navigationOptions = {
    headerShown: false,
  };

  constructor(props) {
    super(props);

    this.state = {
      showRealApp: false,
    };
  }

  renderItem = ({ item }) => {
    if (item.image) {
      return (
        <ScrollView style={styles.slideWrapper}>
          <View style={styles.slide}>
            <Text style={styles.title}>{item.title}</Text>
            <Text style={styles.text}>{item.text}</Text>
            <Image style={styles.image} source={item.image} />
          </View>
        </ScrollView>
      );
    }
    return (
      <ScrollView style={styles.slideWrapper}>
        <View style={styles.slide}>
          <Text style={styles.title}>{item.title}</Text>
          <Text style={styles.text}>{item.text}</Text>
        </View>
      </ScrollView>
    );
  };

  onDone = () => {
    this.setState({ showRealApp: true });
  };

  render() {
    const { showRealApp } = this.state;
    if (showRealApp) {
      return (
        <ScrollView style={styles.slideWrapper}>
          <View>
            <Card>
              <Card.Title>COMPANIES TO APPLY FOR</Card.Title>
              <Card.Divider />
              {validCompanies.map((u, i) => {
                return (
                  <View key={i} style={styles.user}>
                    <Text style={styles.name}>{u.company}</Text>
                    <Card.Divider />
                  </View>
                );
              })}
              <Card.Title>COMPANIES NOT TO APPLY FOR</Card.Title>
              <Card.Divider />
              {inValidCompanies.map((u, i) => {
                return (
                  <View key={i} style={styles.user}>
                    <Text style={styles.name}>{u.company}</Text>
                    <Card.Divider />
                  </View>
                );
              })}
            </Card>
          </View>
        </ScrollView>
      );
    }
    return <AppIntroSlider renderItem={this.renderItem} data={slides} onDone={this.onDone} />;
  }
}
