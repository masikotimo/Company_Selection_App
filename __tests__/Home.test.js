import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import configureMockStore from 'redux-mock-store';
import Home from '../src/screens/Home';

const mockedStore = configureMockStore()({});
const testProps = createTestProps({});

describe('Home', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Home {...testProps} store={mockedStore} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
