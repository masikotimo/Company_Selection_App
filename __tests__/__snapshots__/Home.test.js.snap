// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<AppIntroSlider
  activeDotStyle={
    Object {
      "backgroundColor": "rgba(255, 255, 255, .9)",
    }
  }
  bottomButton={false}
  data={
    Array [
      Object {
        "image": 1,
        "key": 1,
        "text": "Hello I am Timothy Masiko.
Thank you for giving me a chance ",
        "title": "Introduction",
      },
      Object {
        "key": 2,
        "text": "There consists of an Input file in the source code that takes in input inform of a string.

Each line of the string must be in such a format below 

======
\\"Company A\\" requires an apartment or house, and property insurance
======

  The company name must be in quotation marks

A company should not be repeated ",
        "title": "Format Of Input",
      },
      Object {
        "key": 3,
        "text": "Lets dive right in",
        "title": "Alright enough with the Intro ",
      },
    ]
  }
  doneLabel="Done"
  dotClickEnabled={true}
  dotStyle={
    Object {
      "backgroundColor": "rgba(0, 0, 0, .2)",
    }
  }
  nextLabel="Next"
  onDone={[Function]}
  prevLabel="Back"
  renderItem={[Function]}
  showDoneButton={true}
  showNextButton={true}
  showPrevButton={false}
  showSkipButton={false}
  skipLabel="Skip"
/>
`;
